sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareOperation.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.TextAnalytics.Models;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary> The HealthcareOperation class for LRO. </summary>
+    public class HealthcareOperation : Operation<RecognizeHealthcareEntitiesResultCollection>
+    {
+        /// <summary>Provides communication with the Text Analytics Azure Cognitive Service through its REST API.</summary>
+        private readonly TextAnalyticsRestClient _serviceClient;
+
+        /// <summary>Provides tools for exception creation in case of failure.</summary>
+        private readonly ClientDiagnostics _diagnostics;
+
+        /// <summary>
+        /// Provides the input to be part of HealthcareOperation class
+        /// </summary>
+        private readonly IDictionary<string, int> _idToIndexMap;
+
+        /// <summary>
+        /// Gets an ID representing the operation that can be used to poll for the status
+        /// of the long-running operation.
+        /// </summary>
+        public override string Id { get; }
+
+        /// <summary>
+        /// next link string for pagination
+        /// </summary>
+        public string NextLink { get; set; }
```

Comment:
this shouldn't be public. We don't want to expose this to the user as they shouldn't be in charge of doing the pagination